### PR TITLE
feat: lsp completions for ```import new ...```

### PR DESCRIPTION
**Pull request by Daniel-Anker-Hermansen**
related to #4554

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala [hunk 1]
```diff
@@ -848,6 +849,70 @@ object CompletionProvider {
...
+      }
+      case _ => Nil
+    }
+  }
+
+  /**
+    * Converts a Java Class Object into a string representing the type in flix syntax. 
+    * I.e. java.lang.String => String, byte => Int8, java.lang.Object[] => Array[##java.lang.Object, false].
+    */
+  private def convertJavaClassToFlixType(clazz: Class[_ <: Object]): String = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala:896]: @mlutze Is there some better function Daniel can reuse?

**Comment by magnus-madsen** [comment]
!!!jar

**Comment by Daniel-Anker-Hermansen** [comment]
image
image

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala [hunk 2]
```diff
@@ -848,6 +849,70 @@ object CompletionProvider {
...
+              // Prioritise constructors with fewer parameters.
+              sortText = Priority.high(s"${constructor.getParameterCount()}$label"),
+              textEdit = TextEdit(context.range, replace),
+              documentation = None,
+              insertTextFormat = InsertTextFormat.Snippet,
+              kind = CompletionItemKind.Constructor)
+          })
+        }
+        catch {
+          case _: Throwable => Nil
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala:885]: I don't think we should catch Throwable, we should be more precise.
Also, add a comment to explain what this case corresponds to (i.e. the class was not found).

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala [hunk 3]
```diff
@@ -848,6 +849,70 @@ object CompletionProvider {
...
+        case "short" => "Int16"
+        case "int" => "Int32"
+        case "long" => "Int64"
+        case "float" => "Float32"
+        case "double" => "Float64"
+        case "boolean" => "Bool"
+        case "char" => "Char"
+        case "java.lang.String" => "String"
+        case other => s"##$other"
+      }
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala:912]: Step 1: Extend to include stuff from `Resolver.scala`.
Step 2: @mlutze and @magnus-madsen figures out where to put this function in the compiler.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala [hunk 4]
```diff
@@ -848,6 +849,70 @@ object CompletionProvider {
...
+    }
+  }
+
+  /**
+    * Converts a Java Class Object into a string representing the type in flix syntax. 
+    * I.e. java.lang.String => String, byte => Int8, java.lang.Object[] => Array[##java.lang.Object, false].
+    */
+  private def convertJavaClassToFlixType(clazz: Class[_ <: Object]): String = {
+    if (clazz.isArray()) {
+      s"Array[${convertJavaClassToFlixType(clazz.getComponentType())}, false]"
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala:898]: false -> Static

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**